core/imds/src/main/java/software/amazon/awssdk/imds/Ec2Metadata.java

```diff
@@ -29,4 +34,137 @@ public interface Ec2Metadata {
+     * @return The endpoint mode of IMDS.Supported values include IPv4 and IPv6.
+     */
+    String endpointMode();
+
+    /**
+     * @return The output stream for debugging.
+     */
+    String httpDebugOutput();
+
+    /**
+     * @return The number of seconds to sleep in-between retries.
+     */
+    Integer backoff();
+
+    /**
+     * @return The SdkHttpClient instance to make the http requests.
+     */
+    SdkHttpClient httpClient();
+
+    /**
+     * @return The Builder Object consisting all the fields.
+     */
+    Ec2Metadata.Builder toBuilder();
+
+    /**
+     * Create an {@code Ec2Metadata} using the default values.
+     */
+    static Ec2Metadata create() {
+        return builder().build();
+    }
+
+    /**
+     * Creates a default builder for {@link Ec2Metadata}.
+     */
+    static Ec2Metadata.Builder builder() {
+        return DefaultEc2Metadata.builder();
+    }
+
+    /**
+     * The builder definition for a {@link Ec2Metadata}.
+     */
+    interface Builder  {
+
+        /**
+         * Define the retry policy which includes the number of retry attempts for any failed request.
+         *
+         * @param retryPolicy The retry policy which includes the number of retry attempts for any failed request.
+         * @return Returns a reference to this builder
+         */
+        Builder retryPolicy(RetryPolicy retryPolicy);
+
+        /**
+         * Define the endpoint of IMDS.
+         *
+         * @param endpoint The endpoint of IMDS.
+         * @return Returns a reference to this builder
+         */
+        Builder endpoint(URI endpoint);
+
+        /**
+         * Define the port for the endpoint.
+         *
+         * @param port The port for the endpoint.
+         * @return Returns a reference to this builder
+         */
+        Builder port(Integer port);
+
+        /**
+         * Define the Time to live (TTL) of the token.
+         *
+         * @param tokenTtl The Time to live (TTL) of the token.
+         * @return Returns a reference to this builder
+         */
+        Builder tokenTtl(Duration tokenTtl);
+
+        /**
+         * Define the endpoint mode of IMDS.Supported values include IPv4 and IPv6.
+         *
+         * @param endpointMode The endpoint mode of IMDS.Supported values include IPv4 and IPv6.
+         * @return Returns a reference to this builder
+         */
+        Builder endpointMode(String endpointMode);
+
+        /**
+         * Define the output stream for debugging.
+         *
+         * @param httpDebugOutput TThe output stream for debugging.
+         * @return Returns a reference to this builder
+         */
+        Builder httpDebugOutput(String httpDebugOutput);
+
+        /**
+         * Define the number of seconds to sleep in-between retries.
+         *
+         * @param backoff The number of seconds to sleep in-between retries.
+         * @return Returns a reference to this builder
+         */
+        Builder backoff(Integer backoff);
```

Comment:
Isn't this included in the retryPolicy?

Reply:
+1 we should remove this

Reply:
The SEP mentions that the if the user mentions the backoff , it should be an override.  Would it be better to have a field for it as it would be convenient for user ?